cranelift/assembler-x64/src/custom.rs

```diff
@@ -83,6 +86,20 @@ pub mod display {
     lock!(lock_xorw_mr);
     lock!(lock_xorl_mr);
     lock!(lock_xorq_mr);
+
+    pub fn vcvtpd2ps_a<R: Registers>(inst: &inst::vcvtpd2ps_a<R>) -> String {
+        match inst.xmm_m128 {
+            XmmMem::Xmm(_) => "vcvtpd2ps".to_string(),
+            XmmMem::Mem(_) => "vcvtpd2psx".to_string(),
+        }
+    }
+
+    pub fn vcvttpd2dq_a<R: Registers>(inst: &inst::vcvttpd2dq_a<R>) -> String {
+        match inst.xmm_m128 {
+            XmmMem::Xmm(_) => "vcvttpd2dq".to_string(),
+            XmmMem::Mem(_) => "vcvttpd2dqx".to_string(),
```

Comment:
Huh, weird... 🤷

Reply:
I think this is to disambiguate between

```
VEX.128.66.0F.WIG E6 /r
VCVTTPD2DQ xmm1, xmm2/m128
```

and

```
VEX.256.66.0F.WIG E6 /r
VCVTTPD2DQ xmm1, ymm2/m256
```

where in the memory-to-register case the att-style assembler, without the `x` suffix, would look like `vcvttpd2dq (%rax), %xmm1` and it's ambiguous as to which instruction you want. I think the `x` and `y` suffixes are used to disambiguate at the assembler level. Now if we used intel syntax...